FSW/controllers/detumbling_controller.py

```diff
@@ -0,0 +1,453 @@
+import numpy as np
+from actuators.magnetorquer import Magnetorquer
+from world.math.quaternions import *
+
+Z = np.block([
+    [np.eye(3),       np.zeros((3,3))],
+    [np.zeros((3,3)), np.zeros((3,3))],
+])
+
+
+def convert_to_skew_symmetric(
+        vec: np.ndarray,
+    ) -> np.ndarray:
+        return np.array([
+            [0.0, -vec[2], vec[1]],
+            [vec[2], 0.0, -vec[0]],
+            [-vec[1], vec[0], 0.0],
+        ])
+
+
+class NonMonotonicController():
+    def __init__(
+        self,
+        inertia_tensor: np.ndarray,
+        time_step: float,
+        alpha=100.0,
+        beta=1.0,
+        dipole_moment_lower_bound=np.zeros(3),
+        dipole_moment_upper_bound=np.finfo(np.float64).max,
+    ) -> None:
+        self.J = inertia_tensor
+        self.dt = time_step
+        self.alpha = alpha
+        self.beta = beta
+        self.lbm = dipole_moment_lower_bound
+        self.ubm = dipole_moment_upper_bound
+
+    def get_dipole_moment_command(
+        self,
+        magnetic_field_in_eci_frame: np.ndarray,
```

Comment:
FSW does not get to dictate which frame the arguments are provided in. It only receives sensor data and we cannot add a pre-processing step. So we should assume all quantites are in the body frame (except some sensors like star-tracker that operate in ECI)

Reply:
this is not supposed to be the actual fsw, only modelling what it does. but yeah that there is not correct

Reply:
this is part of the nonmonotonic detumbling controller which we won't be using